spec/services/draw_activator_spec.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+RSpec.describe DrawActivator do
+  # TODO: extract this into a shared example; if we're going to use this pattern
+  # across all of our service objects we should have a way to easily implement
+  # it for all of them. Should be pretty easy - all we need to pass to the
+  # shared example is the method we want to call and the params hash
+  describe '.start' do
+    it 'calls :start on an instance of DrawStarter' do
+      draw = instance_spy('Draw')
+      draw_activator = create_mock_draw_activator(draw: draw)
+      described_class.activate(draw: draw)
+
+      expect(draw_activator).to have_received(:activate)
+    end
+  end
+
+  describe '#activate' do
+    it 'checks to make sure that the draw is a draft' do
+      draw = instance_spy('draw', draft?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to include('Draw must be a draft.')
+    end
+
+    it 'checks to make sure that the draw has at least one student' do
+      draw = instance_spy('draw', has_students?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to \
+        include('Draw must have at least one student.')
+    end
+
+    it 'checks to make sure that there are enough beds for students' do
+      draw = instance_spy('draw', has_enough_beds?: false)
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:msg][:error]).to \
+        include('Draw must have at least one bed per student')
+    end
+
+    it 'updates the status of the draw to pre_lottery' do
+      # draw = create_mock_draw
+      draw = create_draw_data
+
+      result = described_class.activate(draw: draw)
+
+      expect(result[:obj]).to be_pre_lottery
+    end
+  end
+
+  def create_mock_draw_activator(param_hash)
+    instance_spy('DrawActivator').tap do |draw_activator|
+      allow(described_class).to receive(:new).with(param_hash)
+        .and_return(draw_activator)
+    end
+  end
+
+  # NOTE: I don't think that instance spies work with ActiveRecord models. I ran
+  # into issues with stubbing the `pre_lottery?` method when the enum value was
+  # 'pre-lottery' (dashes are problematic in Ruby symbols) - but there is no
+  # verification of the fact that the `:pre_lottery?` method is a valid method
+  # on the Draw class, e.g. that the 'pre_lottery' value is a valid enum value.
+  # This kinda makes sense since a lot of this is accomplished through
+  # `method_missing`, but it's def something to keep in mind. The problem is
+  # that we can write a test that would pass against broken code since it
+  # doesn't check for enum validity. I tried switching to FactoryGirl's
+  # `build_stubbed` but I ran into major issues with data being persisted or
+  # being unable to generate a valid draw for the service object without
+  # persisting all the things. Thoughts?
```

Comment:
Thanks for confirming my suspicions and validating that I wasn't being entirely stupid. I'm open to implementing the `Mocker` again if we feel we need it - I think we've been ok without the additional complexity and just manually dealing with `instance_spy`s, but we can come back to this if it comes up.